File: src/main/java/net/ccbluex/liquidbounce/ui/client/gui/clickgui/utils/render/GLUtil.java
/*
 * FDPClient Hacked Client
 * A free open source mixin-based injection hacked client for Minecraft using Minecraft Forge by LiquidBounce.
 * https://github.com/SkidderMC/FDPClient/
 */
package net.ccbluex.liquidbounce.ui.client.gui.clickgui.utils.render;

import net.minecraft.client.renderer.GlStateManager;

import static org.lwjgl.opengl.GL11.*;

public class GLUtil {

    public static void render(int mode, Runnable render){
        glBegin(mode);
        render.run();
        glEnd();
    }

    public static void setup2DRendering(Runnable f) {
        glEnable(GL_BLEND);
        glBlendFunc(GL_SRC_ALPHA, GL_ONE_MINUS_SRC_ALPHA);
        glDisable(GL_TEXTURE_2D);
        f.run();
        glEnable(GL_TEXTURE_2D);
        GlStateManager.disableBlend();
    }

    public static void rotate(float x, float y, float rotate, Runnable f) {
        GlStateManager.pushMatrix();
        GlStateManager.translate(x, y, 0);
        GlStateManager.rotate(rotate, 0, 0, -1);
        GlStateManager.translate(-x, -y, 0);
        f.run();
        GlStateManager.popMatrix();
    }

}

File: src/main/java/net/ccbluex/liquidbounce/ui/client/clickgui/style/styles/nlclickgui/GLUtil.kt
package net.ccbluex.liquidbounce.ui.client.clickgui.style.styles.nlclickgui

import net.minecraft.client.renderer.GlStateManager
import org.lwjgl.opengl.GL11

object GLUtil {
    fun render(mode: Int, render: Runnable) {
        GL11.glBegin(mode)
        render.run()
        GL11.glEnd()
    }

    fun setup2DRendering(f: Runnable) {
        GL11.glEnable(GL11.GL_BLEND)
        GL11.glBlendFunc(GL11.GL_SRC_ALPHA, GL11.GL_ONE_MINUS_SRC_ALPHA)
        GL11.glDisable(GL11.GL_TEXTURE_2D)
        f.run()
        GL11.glEnable(GL11.GL_TEXTURE_2D)
        GlStateManager.disableBlend()
    }

    fun rotate(x: Float, y: Float, rotate: Float, f: Runnable) {
        GlStateManager.pushMatrix()
        GlStateManager.translate(x, y, 0f)
        GlStateManager.rotate(rotate, 0f, 0f, -1f)
        GlStateManager.translate(-x, -y, 0f)
        f.run()
        GlStateManager.popMatrix()
    }
}